src/openforms/forms/api/viewsets.py

```diff
@@ -294,7 +293,94 @@ def export(self, request, *args, **kwargs):
+            type=str,
+            description=_("Either a UUID4 or a slug identifying the form."),
+        )
+    ]
+)
+@extend_schema_view(
+    create=extend_schema(
+        summary=_("Save form version"),
+        tags=["forms"],
+        responses={status.HTTP_201_CREATED: FormVersionSerializer},
+    ),
+    restore=extend_schema(
+        summary=_("Restore form version"),
+        tags=["forms"],
+        parameters=[
+            OpenApiParameter(
+                name="form_uuid_or_slug",
+                location=OpenApiParameter.PATH,
+                type=str,
+                description=_("Either a UUID4 or a slug identifying the form."),
+            ),
+            OpenApiParameter(
+                name="uuid",
+                location=OpenApiParameter.PATH,
+                type=str,
+                description=_("The uuid of the form version"),
+            ),
+        ],
+        responses={status.HTTP_201_CREATED: ""},
+    ),
+    list=extend_schema(
+        summary=_("List form versions"),
+        tags=["forms"],
+        parameters=[
+            OpenApiParameter(
+                name="form_uuid_or_slug",
+                location=OpenApiParameter.PATH,
+                type=str,
+                description=_("Either a UUID4 or a slug identifying the form."),
+            )
+        ],
+        request=FormVersionSerializer,
+        responses={status.HTTP_200_OK: FormVersionSerializer(many=True)},
+    ),
+)
+class FormVersionViewSet(NestedViewSetMixin, viewsets.ViewSet):
+    serializer_class = FormVersionSerializer
+    queryset = FormVersion.objects.all()
+    permission_classes = [IsStaffOrReadOnly]
+    lookup_field = "uuid"
+    parent_lookup_kwargs = {"form_uuid_or_slug": "form__uuid"}
+
+    def create(self, request, *args, **kwargs):
+        """Save the current version of the form so that it can later be retrieved"""
+
+        form = get_object_or_404(Form, uuid=self.kwargs["form_uuid_or_slug"])
+
+        form_json = form_to_json(form.id)
+        form_version = FormVersion.objects.create(form=form, export_blob=form_json)
+
+        serializer = self.serializer_class(instance=form_version)
+        return Response(status=status.HTTP_201_CREATED, data=serializer.data)
+
+    @action(detail=True, methods=["post"])
+    def restore(self, request, *args, **kwargs):
+        """Restore a previously saved version of a form."""
+        form = get_object_or_404(Form, uuid=self.kwargs["form_uuid_or_slug"])
+        form_version = get_object_or_404(FormVersion, uuid=self.kwargs["uuid"])
+
+        form.restore_old_version(form_version.uuid)
+
+        return Response(status=status.HTTP_201_CREATED)
+
+    def list(self, request, *args, **kwargs):
+        """List all the versions of a form"""
+        form = get_object_or_404(Form, uuid=self.kwargs["form_uuid_or_slug"])
+
+        form_version_serializer = FormVersionSerializer(
+            instance=FormVersion.objects.filter(form=form), many=True
+        )
```

Comment:
I feel like this should just be using the `ListModelMixin` combined with filtering the queryset from the `NestedViewSetMixin`.